base/tuple.jl

```diff
@@ -1,5 +1,12 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
+# Document NTuple here where we have everything needed for the doc system
+"""
+    NTuple{N, T}
```

Comment:
Don't know. It is after all just a `typealias` however it is quite frequently used so maybe..